### ISSUE TITLE
Sourcehold under Linux

### ISSUE THREAD

**Issue by nitramr** [sourcehold/Sourcehold#20]
Hi,

is the launcher made to run under Linux? At this moment I could compile the Sourcehold launcher, but I run into an error: "There appears to be no settings.ini in your installation path!"

It seems like that the settings.ini is auto generated by installation of original game. I have the original game, but I cannot run the original setup on Linux. So I never will get the settings.ini file.
For data folder I used unshield (https://github.com/twogood/unshield) to extract the cab files from original game.

Is it possible that the Sourcehold launcher creates the settings file? Thanks.

**Comment by metalvoidzz** [comment]
Hi, the settings.ini file is a way to permanently add command-line options. That includes ones which weren't present in the original game. Because of that, Sourcehold uses both the Stronghold.cfg from the original game and the (optional) settings.ini.

I added that feature recently and didn't add a default config file yet, but the game should run without one (the "There appears to be no settings.ini in your installation path!" message is just a warning). The default command-line settings should work fine, so please let me know if you run into an error regardless.

**Comment by nitramr** [comment]
Hi,
thanks for the fast reply. Since it is not connected to the missing settings.ini file I made a reseach why it can't start on my Ubuntu 19.04 system.

I started the game via console and it will exit with "memory access error".

In the source code I added some log messages to figure out when it stop working.
Looks like the game have problems to load the texture atlas in startup.cpp:
```
UIState Startup::Begin()
{
...
    auto font = GetGm1("gm/font_stronghold_aa.gm1")->GetTextureAtlas();
...
```
I'm using the game assets from 2001.
I will let you know if I have more infos, perhaps I can fix it by myself. :)
